src/squeue.rs

```diff
@@ -240,14 +246,77 @@ impl SubmissionQueue<'_> {
         }
 
         for Entry(entry) in entries {
-            *self
+            *self.next_sqe() = *entry;
+            self.move_forward();
+        }
+
+        Ok(())
+    }
+
+    /// Attempts to push an opcode into the submission queue.
+    /// If the queue is full, an error is returned.
+    ///
+    /// # Safety
+    ///
+    /// Developers must ensure that parameters of the opcode (such as buffer) are valid and will
+    /// be valid for the entire duration of the operation, otherwise it may cause memory problems.
+    #[inline]
+    pub unsafe fn push_command<'a, T: PrepareSQE>(
+        &'a mut self,
+        opcode: &T,
+        options: Option<&OptionValues>,
+    ) -> Result<(), PushError> {
+        if !self.is_full() {
+            let sqe = self.next_sqe();
+            opcode.prepare(sqe, options);
+            self.move_forward();
+            Ok(())
+        } else {
+            Err(PushError)
+        }
+    }
+
+    /// Attempts to push several opcodes into the queue.
+    /// If the queue does not have space for all of the entries, an error is returned.
+    ///
+    /// # Safety
+    ///
+    /// Developers must ensure that parameters of all the entries (such as buffer) are valid and
+    /// will be valid for the entire duration of the operation, otherwise it may cause memory
+    /// problems.
+    #[cfg(feature = "unstable")]
+    #[inline]
+    pub unsafe fn push_commands<'a, T: PrepareSQE>(
+        &'a mut self,
+        ops: &[(T, Option<&OptionValues>)],
+    ) -> Result<(), PushError> {
+        if self.capacity() - self.len() < ops.len() {
+            return Err(PushError);
+        }
+
+        for (opcode, options) in ops {
+            let sqe = self.next_sqe();
+            opcode.prepare(sqe, *options);
+            self.move_forward();
+        }
+
+        Ok(())
+    }
+
+    #[inline]
+    fn next_sqe(&mut self) -> &mut sys::io_uring_sqe {
```

Comment:
This method and the following `move_forward` should be unsafe, because it may return entry being used by kernel and make kernel use the uninitialized entry.

Reply:
Fixed